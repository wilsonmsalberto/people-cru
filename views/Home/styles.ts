import styled from '@emotion/styled';

export const Wrapper = styled.section`
  margin-top: 4rem;
  display: flex;
  justify-content: space-between;
  align-items: center;
`;

export const TitleWrapper = styled.div`
  display: flex;
  justify-content: space-between;
  align-items: flex-end;
`;

export const Title = styled.h1`
  margin-right: 0.8rem;
  font-weight: 500;
  font-size: 2.6rem;
  line-height: 3.1rem;
`;

export const SubTitle = styled.span`
  font-weight: 500;
  font-size: 1.3rem;
  line-height: 2.1rem;
  color: ${({ theme }) => theme.secondary};
`;
