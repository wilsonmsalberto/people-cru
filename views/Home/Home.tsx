import { useContext } from 'react';
import { useRouter } from 'next/router';

// Context
import { PeopleListContext, PeopleListProvider } from 'context/PeopleList';

// Components
import Button from 'ui/components/Button';
import PeopleList from 'ui/blocks/PeopleList/index';

// Styles
import { SubTitle, Title, TitleWrapper, Wrapper } from './styles';

const Home = (): React.ReactElement => {
    const { resultsAmount } = useContext(PeopleListContext);
    const router = useRouter();

    return (
        <>
            <Wrapper data-testid="home">
                <TitleWrapper>
                    <Title>People</Title>
                    <SubTitle>{ resultsAmount } employees</SubTitle>
                </TitleWrapper>

                <Button
                    type="submit"
                    onClick={ (): Promise<any> => router.push('/new') }
                >
                    Add employee
                </Button>
            </Wrapper>

            <PeopleList />
        </>
    );
};

const withContextHome = (): React.ReactElement => (
    <PeopleListProvider>
        <Home />
    </PeopleListProvider>
);

export default withContextHome;
