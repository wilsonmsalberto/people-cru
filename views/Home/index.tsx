import React, { useContext } from 'react';
import { useRouter } from 'next/router';

import { Wrapper, TitleWrapper, Title, SubTitle } from './styles';

import { PeopleListContext, PeopleListProvider } from 'context/PeopleList';

import Button from 'ui/components/Button';
import PeopleList from 'ui/blocks/PeopleList/index';

const Home = (): React.ReactElement => {
  const { resultsAmount } = useContext(PeopleListContext);
  const router = useRouter();

  const handleAddNewClick = (): Promise<any> => router.push('/new');

  return (
    <>
      <Wrapper>
        <TitleWrapper>
          <Title>People</Title>
          <SubTitle>{resultsAmount} employees</SubTitle>
        </TitleWrapper>

        <Button btnStyle="primary" type="submit" onClick={handleAddNewClick}>
          Add employee
        </Button>
      </Wrapper>
      <PeopleList />
    </>
  );
};

const withContextHome = (): React.ReactElement => (
  <PeopleListProvider>
    <Home />
  </PeopleListProvider>
);

export default withContextHome;
