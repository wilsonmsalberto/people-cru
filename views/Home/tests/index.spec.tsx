import React from 'react';
import { render } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
import { matchers } from 'jest-emotion';

import Home from '..';

expect.extend(matchers);

describe('Home Page', () => {
  describe('Rendering', () => {
    it('should render properly', () => {
      const { asFragment } = render(<Home />);

      expect(asFragment()).toMatchSnapshot();
    });
  });
});
