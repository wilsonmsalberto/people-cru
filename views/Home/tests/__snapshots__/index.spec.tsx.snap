// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home Page Rendering should render properly 1`] = `
<DocumentFragment>
  <section
    class="css-11s1md0"
  >
    <div
      class="css-6fqrth"
    >
      <h1
        class="css-16i1488"
      >
        People
      </h1>
      <span
        class="css-9lpvpw"
      >
        20 employees
      </span>
    </div>
    <button
      class="css-1f5124n"
      type="submit"
    >
      Add employee
    </button>
  </section>
  <table
    class="css-148mg3b"
  >
    <thead
      class="css-1kse8ce"
    >
      <tr>
        <th>
          Employee
        </th>
        <th>
          Job Title
        </th>
        <th>
          Country
        </th>
        <th
          colspan="2"
        >
          Salary
        </th>
      </tr>
    </thead>
    <tbody>
      <tr
        class="css-1iaf3li"
      >
        <td>
          <p
            class="css-vdtd8u"
          >
            Justin Colvill
          </p>
          <span
            class="css-1tvovr2"
          >
            03/11/1938
          </span>
        </td>
        <td>
          Sales Representative
        </td>
        <td>
          Greece
        </td>
        <td
          class="css-epn6zh"
        >
          €2181,66 
          <span>
            per month
          </span>
        </td>
        <td
          class="css-889chh"
        >
          <button
            class="css-1fa7ku2"
            type="button"
          >
            Edit
          </button>
        </td>
      </tr>
      <tr
        class="css-1iaf3li"
      >
        <td>
          <p
            class="css-vdtd8u"
          >
            Ranique Bern
          </p>
          <span
            class="css-1tvovr2"
          >
            07/10/1941
          </span>
        </td>
        <td>
          Geological Engineer
        </td>
        <td>
          China
        </td>
        <td
          class="css-epn6zh"
        >
          €1686,04 
          <span>
            per month
          </span>
        </td>
        <td
          class="css-889chh"
        >
          <button
            class="css-1fa7ku2"
            type="button"
          >
            Edit
          </button>
        </td>
      </tr>
      <tr
        class="css-1iaf3li"
      >
        <td>
          <p
            class="css-vdtd8u"
          >
            Vonnie Allix
          </p>
          <span
            class="css-1tvovr2"
          >
            07/06/1925
          </span>
        </td>
        <td>
          Senior Editor
        </td>
        <td>
          Indonesia
        </td>
        <td
          class="css-epn6zh"
        >
          €1024,96 
          <span>
            per month
          </span>
        </td>
        <td
          class="css-889chh"
        >
          <button
            class="css-1fa7ku2"
            type="button"
          >
            Edit
          </button>
        </td>
      </tr>
      <tr
        class="css-1iaf3li"
      >
        <td>
          <p
            class="css-vdtd8u"
          >
            Abagael Thames
          </p>
          <span
            class="css-1tvovr2"
          >
            19/04/1928
          </span>
        </td>
        <td>
          Quality Control Specialist
        </td>
        <td>
          China
        </td>
        <td
          class="css-epn6zh"
        >
          €1452,00 
          <span>
            per month
          </span>
        </td>
        <td
          class="css-889chh"
        >
          <button
            class="css-1fa7ku2"
            type="button"
          >
            Edit
          </button>
        </td>
      </tr>
      <tr
        class="css-1iaf3li"
      >
        <td>
          <p
            class="css-vdtd8u"
          >
            Berkeley Hurdwell
          </p>
          <span
            class="css-1tvovr2"
          >
            07/03/1924
          </span>
        </td>
        <td>
          Environmental Specialist
        </td>
        <td>
          Indonesia
        </td>
        <td
          class="css-epn6zh"
        >
          €2208,47 
          <span>
            per month
          </span>
        </td>
        <td
          class="css-889chh"
        >
          <button
            class="css-1fa7ku2"
            type="button"
          >
            Edit
          </button>
        </td>
      </tr>
      <tr
        class="css-1iaf3li"
      >
        <td>
          <p
            class="css-vdtd8u"
          >
            Osborne Sutherby
          </p>
          <span
            class="css-1tvovr2"
          >
            17/01/1921
          </span>
        </td>
        <td>
          Nurse Practicioner
        </td>
        <td>
          France
        </td>
        <td
          class="css-epn6zh"
        >
          €4058,49 
          <span>
            per month
          </span>
        </td>
        <td
          class="css-889chh"
        >
          <button
            class="css-1fa7ku2"
            type="button"
          >
            Edit
          </button>
        </td>
      </tr>
      <tr
        class="css-1iaf3li"
      >
        <td>
          <p
            class="css-vdtd8u"
          >
            Adelina Dymott
          </p>
          <span
            class="css-1tvovr2"
          >
            17/06/1922
          </span>
        </td>
        <td>
          Web Developer I
        </td>
        <td>
          Serbia
        </td>
        <td
          class="css-epn6zh"
        >
          €4576,20 
          <span>
            per month
          </span>
        </td>
        <td
          class="css-889chh"
        >
          <button
            class="css-1fa7ku2"
            type="button"
          >
            Edit
          </button>
        </td>
      </tr>
      <tr
        class="css-1iaf3li"
      >
        <td>
          <p
            class="css-vdtd8u"
          >
            Almeta Hazeley
          </p>
          <span
            class="css-1tvovr2"
          >
            16/05/1941
          </span>
        </td>
        <td>
          Desktop Support Technician
        </td>
        <td>
          Indonesia
        </td>
        <td
          class="css-epn6zh"
        >
          €1458,96 
          <span>
            per month
          </span>
        </td>
        <td
          class="css-889chh"
        >
          <button
            class="css-1fa7ku2"
            type="button"
          >
            Edit
          </button>
        </td>
      </tr>
      <tr
        class="css-1iaf3li"
      >
        <td>
          <p
            class="css-vdtd8u"
          >
            Beth Briereton
          </p>
          <span
            class="css-1tvovr2"
          >
            01/10/1927
          </span>
        </td>
        <td>
          Actuary
        </td>
        <td>
          Pakistan
        </td>
        <td
          class="css-epn6zh"
        >
          €3035,36 
          <span>
            per month
          </span>
        </td>
        <td
          class="css-889chh"
        >
          <button
            class="css-1fa7ku2"
            type="button"
          >
            Edit
          </button>
        </td>
      </tr>
      <tr
        class="css-1iaf3li"
      >
        <td>
          <p
            class="css-vdtd8u"
          >
            Garfield Ten
          </p>
          <span
            class="css-1tvovr2"
          >
            15/05/1926
          </span>
        </td>
        <td>
          Analog Circuit Design manager
        </td>
        <td>
          Indonesia
        </td>
        <td
          class="css-epn6zh"
        >
          €3046,85 
          <span>
            per month
          </span>
        </td>
        <td
          class="css-889chh"
        >
          <button
            class="css-1fa7ku2"
            type="button"
          >
            Edit
          </button>
        </td>
      </tr>
      <tr
        class="css-1iaf3li"
      >
        <td>
          <p
            class="css-vdtd8u"
          >
            Byrann Lippo
          </p>
          <span
            class="css-1tvovr2"
          >
            27/12/1932
          </span>
        </td>
        <td>
          Marketing Assistant
        </td>
        <td>
          Croatia
        </td>
        <td
          class="css-epn6zh"
        >
          €3224,58 
          <span>
            per month
          </span>
        </td>
        <td
          class="css-889chh"
        >
          <button
            class="css-1fa7ku2"
            type="button"
          >
            Edit
          </button>
        </td>
      </tr>
      <tr
        class="css-1iaf3li"
      >
        <td>
          <p
            class="css-vdtd8u"
          >
            Shina Woosnam
          </p>
          <span
            class="css-1tvovr2"
          >
            25/01/1934
          </span>
        </td>
        <td>
          Account Coordinator
        </td>
        <td>
          Peru
        </td>
        <td
          class="css-epn6zh"
        >
          €3262,01 
          <span>
            per month
          </span>
        </td>
        <td
          class="css-889chh"
        >
          <button
            class="css-1fa7ku2"
            type="button"
          >
            Edit
          </button>
        </td>
      </tr>
      <tr
        class="css-1iaf3li"
      >
        <td>
          <p
            class="css-vdtd8u"
          >
            Joann Iacovazzi
          </p>
          <span
            class="css-1tvovr2"
          >
            27/04/1943
          </span>
        </td>
        <td>
          Geological Engineer
        </td>
        <td>
          Indonesia
        </td>
        <td
          class="css-epn6zh"
        >
          €3234,48 
          <span>
            per month
          </span>
        </td>
        <td
          class="css-889chh"
        >
          <button
            class="css-1fa7ku2"
            type="button"
          >
            Edit
          </button>
        </td>
      </tr>
      <tr
        class="css-1iaf3li"
      >
        <td>
          <p
            class="css-vdtd8u"
          >
            Elisabet De
          </p>
          <span
            class="css-1tvovr2"
          >
            03/04/1941
          </span>
        </td>
        <td>
          Associate Professor
        </td>
        <td>
          Poland
        </td>
        <td
          class="css-epn6zh"
        >
          €2075,11 
          <span>
            per month
          </span>
        </td>
        <td
          class="css-889chh"
        >
          <button
            class="css-1fa7ku2"
            type="button"
          >
            Edit
          </button>
        </td>
      </tr>
      <tr
        class="css-1iaf3li"
      >
        <td>
          <p
            class="css-vdtd8u"
          >
            Kenny Attwoull
          </p>
          <span
            class="css-1tvovr2"
          >
            16/02/1940
          </span>
        </td>
        <td>
          Geologist III
        </td>
        <td>
          China
        </td>
        <td
          class="css-epn6zh"
        >
          €4741,50 
          <span>
            per month
          </span>
        </td>
        <td
          class="css-889chh"
        >
          <button
            class="css-1fa7ku2"
            type="button"
          >
            Edit
          </button>
        </td>
      </tr>
      <tr
        class="css-1iaf3li"
      >
        <td>
          <p
            class="css-vdtd8u"
          >
            Ralf Mockett
          </p>
          <span
            class="css-1tvovr2"
          >
            27/11/1948
          </span>
        </td>
        <td>
          Pharmacist
        </td>
        <td>
          China
        </td>
        <td
          class="css-epn6zh"
        >
          €1327,68 
          <span>
            per month
          </span>
        </td>
        <td
          class="css-889chh"
        >
          <button
            class="css-1fa7ku2"
            type="button"
          >
            Edit
          </button>
        </td>
      </tr>
      <tr
        class="css-1iaf3li"
      >
        <td>
          <p
            class="css-vdtd8u"
          >
            Gun Kittles
          </p>
          <span
            class="css-1tvovr2"
          >
            05/03/1926
          </span>
        </td>
        <td>
          Director of Sales
        </td>
        <td>
          Portugal
        </td>
        <td
          class="css-epn6zh"
        >
          €4442,08 
          <span>
            per month
          </span>
        </td>
        <td
          class="css-889chh"
        >
          <button
            class="css-1fa7ku2"
            type="button"
          >
            Edit
          </button>
        </td>
      </tr>
      <tr
        class="css-1iaf3li"
      >
        <td>
          <p
            class="css-vdtd8u"
          >
            Viviene Ponde
          </p>
          <span
            class="css-1tvovr2"
          >
            05/02/1929
          </span>
        </td>
        <td>
          Software Consultant
        </td>
        <td>
          Ukraine
        </td>
        <td
          class="css-epn6zh"
        >
          €2386,21 
          <span>
            per month
          </span>
        </td>
        <td
          class="css-889chh"
        >
          <button
            class="css-1fa7ku2"
            type="button"
          >
            Edit
          </button>
        </td>
      </tr>
      <tr
        class="css-1iaf3li"
      >
        <td>
          <p
            class="css-vdtd8u"
          >
            Audi Ashbrook
          </p>
          <span
            class="css-1tvovr2"
          >
            23/02/1936
          </span>
        </td>
        <td>
          GIS Technical Architect
        </td>
        <td>
          Indonesia
        </td>
        <td
          class="css-epn6zh"
        >
          €3346,58 
          <span>
            per month
          </span>
        </td>
        <td
          class="css-889chh"
        >
          <button
            class="css-1fa7ku2"
            type="button"
          >
            Edit
          </button>
        </td>
      </tr>
      <tr
        class="css-1iaf3li"
      >
        <td>
          <p
            class="css-vdtd8u"
          >
            Durand Depport
          </p>
          <span
            class="css-1tvovr2"
          >
            25/09/1927
          </span>
        </td>
        <td>
          Human Resources Assistant I
        </td>
        <td>
          Bulgaria
        </td>
        <td
          class="css-epn6zh"
        >
          €1518,80 
          <span>
            per month
          </span>
        </td>
        <td
          class="css-889chh"
        >
          <button
            class="css-1fa7ku2"
            type="button"
          >
            Edit
          </button>
        </td>
      </tr>
    </tbody>
  </table>
</DocumentFragment>
`;
